clara/parser.py

```diff
@@ -408,6 +408,8 @@ def visit_loop(self, node, init, cond, next, body, do, name, prebody=None):
 
         # Add condition (with new location)
         preloc = self.loc
+        condloc = self.addloc("the condition of the '%s' loop at line %s" % (
+            name, self.getline(node)))
```

Comment:
Why exactly is this necessary? Please do add test cases for any change to better understand and make sure everything is fine.

Reply:
I should revisit code later to answer questions about this PR.

Briefly, in some cases like:
```c
while(i++) {
    j++;
}
```
the body of loop condition (`i++`) ran only once. This PR tries to fix this. (I'm writing the example from my memory, so I'm not so sure about it.)

Reply:
Ah, OK. I see.